fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ExpressionAnalyzer.java

```diff
@@ -650,6 +651,15 @@ public Expression visitComparisonPredicate(ComparisonPredicate cp, ExpressionRew
         // Used to replace expression in ShortCircuit plan
         registerPlaceholderIdToSlot(cp, context, left, right);
         cp = (ComparisonPredicate) cp.withChildren(left, right);
+        if (cp.containsType(SubqueryExpr.class) && getScope().getOuterScope().isPresent()) {
```

Comment:
we should enum all place that comaprison predicate could appear and test them to make should we could process all situation correctly